server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -279,4 +319,14 @@ private VaultAction getVaultAction(String actionString) {
                 actionString,
                 Joiner.on(",").join(VaultAction.plainValues()))));
     }
+
+    private void validatePurgeAction(VaultAction requestedAction) {
+        if (requestedAction != VaultAction.PURGE) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Not 'purge' action")
```

Comment:
it's not what is written in the previous condition

Reply:
Replace this method with more general method
```java
    private void validateAction(Request request, VaultAction... validActions) {
        VaultAction requestedAction = extractVaultAction(request);
        validateAction(requestedAction, validActions);
    }

    private HaltException throwNotSupportedAction(VaultAction requestAction, VaultAction... validActions) {
        return ErrorResponder.builder()
            .statusCode(HttpStatus.BAD_REQUEST_400)
            .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
            .message(String.format("'%s' is not a valid action. Supported values are: %s",
                requestAction.getValue(),
                Joiner.on(", ").join(Stream.of(validActions)
                    .map(action -> String.format("'%s'", action.getValue()))
                    .collect(Guavate.toImmutableList()))))
            .haltError();
    }
```